frontend/rolecall/src/app/cast/cast-drag-and-drop.component.html

```diff
@@ -27,60 +27,106 @@
-                  (click)="decrementDancerCount(i1)">
-                <mat-icon [inline]="true"
-                    class="dancer-count-icon">
-                  remove_circle_outline
-                </mat-icon>
-              </button>
-              <button
-                  mat-icon-button
-                  aria-label="Increment Dancer Count Button"
-                  class="dancer-count-button"
-                  (click)="incrementDancerCount(i1)">
-                <mat-icon [inline]="true"
-                    class="dancer-count-icon">
-                  add_circle_outline
-                </mat-icon>
-              </button>
+              <div class="double-header-cell">
+                <div class="double-header-text">
+                  {{ castPositions[i1].pos.name }}
+                </div>
+                <div class="double-header-buttons">
+                  <div class="button-v-wrapper">
+                    <div class="button-v-container">
+                      <button
+                          mat-icon-button
+                          aria-label="Decrement Dancer Count Button"
+                          class="count-button1"
+                          [disabled]="position.dancerCount < 2"
+                          (click)="decrementDancerCount(i1)">
+                        <mat-icon
+                            [inline]="true"
+                            class="count-icon1">
+                          remove_circle_outline
+                        </mat-icon>
+                      </button>
+                      <button
+                          mat-icon-button
+                          aria-label="Increment Dancer Count Button"
+                          class="count-button2"
+                          (click)="incrementDancerCount(i1)">
+                        <mat-icon
+                            [inline]="true"
+                            class="count-icon2">
+                          add_circle_outline
+                        </mat-icon>
+                      </button>
+                    </div>
+                  </div>  
+                </div>
+              </div>  
             </div>
             <div
-                *ngFor="let header of subCastHeaders"
-                class="cell cellColHeader">
+                *ngFor="let header of subCastHeaders; let i2 = index"
+                class="cell cellColHeader"
+                [class.adjust-last-cast-cell]="i2 === subCastHeaders.length - 1">
+              <ng-container
+                  *ngIf="i2 < subCastHeaders.length - 1; else lastCast">
                 {{ header }}
+              </ng-container>
+              <ng-template #lastCast>
+                <div class="last-cast-cell">
+                  <div class="last-cast-text">
+                    {{ header }}
+                  </div>
+                  <div class="button-h-wrapper">
+                    <div class="button-h-container">
```

Comment:
Why do we need a div inside a div? `<div>`s that only contain another `<div>` should probably all be flattened to just one `<div>` layer with multiple classes if needed. Otherwise it makes the HTML more messy, forces browser to generate unnecessary elements, and also hurts accessibility.

Reply:
Done.

Reply:
Still more work to do on this in the file, but don't want to hold up this PR.